sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/AzureEventHubsAutoConfiguration.java

```diff
@@ -45,7 +48,8 @@ AzureEventHubsProperties azureEventHubsProperties() {
     }
 
     @Bean
-    @ConditionalOnProperty("spring.cloud.azure.eventhubs.connection-string")
+    @ConditionalOnExpression("'${spring.cloud.azure.eventhubs.connection-string:}' != ''")
+    @ConditionalOnMissingBean(value = AzureServiceType.EventHubs.class, parameterizedContainer = ServiceConnectionStringProvider.class)
```

Comment:
Why is this necessary?

Reply:
Not required, but the same support of conn string availability is also useful for Event Hubs configuration.

Reply:
> Not required

Then how about deleting it?

> but the same support of conn string availability is also useful for Event Hubs configuration.

I don't understand this.

Reply:
Communicated offline

Reply:
I think it is because we are injecting a `StaticConnectionStringProvider` bean here. @moarychan, but why we are using conditional on missing bean here?